subt/crash_report.py

```diff
@@ -35,4 +35,60 @@ def update(self):
             assert False, channel  # unsupported channel
         return channel
 
+
+if __name__ == "__main__":
+    import argparse
+    import numpy as np
+    from osgar.lib.serialize import deserialize
+    from osgar.logger import LogReader, LogWriter, lookup_stream_id, lookup_stream_names
+
+    parser = argparse.ArgumentParser("Analyze crash_rgbd data")
+    parser.add_argument('logfile', help='path to logfile')
+    parser.add_argument('--verbose', '-v', help="verbose mode", action='store_true')
+    args = parser.parse_args()
+
+    crash_stream_id = lookup_stream_id(args.logfile, 'black_box.crash_rgbd')
+    pose3d_stream_id = lookup_stream_id(args.logfile, 'fromrospy.pose3d')
+
+    def nearest_pose(poses, pose3d):
+        xyz_arr = np.array([xyz for xyz, quat in poses])
+        xyz_diff = xyz_arr - pose3d[0]
+        dist = np.linalg.norm(xyz_diff, axis=1)
+        index = np.argmin(dist)
+        return index
+
+    stream_names = lookup_stream_names(args.logfile)
+
+    def get_camera_name_id(camera_pose):
+        x, y = camera_pose[0][:2]
+        if y == 0:
+            camera_direction = 'front' if x > 0 else 'rear'
+        elif y < 0:
+            camera_direction = 'right'
+        else:  # y > 0
+            camera_direction = 'left'
```

Comment:
+ rear camera on Freyja?

Do you plan to support K2, where the camera positions change over time?

Reply:
Good point, but I would not worry about it at the moment. I suppose that there will be more changes to solve new needs, and with K2 crash log this can be extended (I guess something like +/-90deg are side cameras and rest is still x based?